server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java

```diff
@@ -323,6 +331,21 @@ public T unremove(String name) {
         return unremovedChild;
     }
 
+    @Override
+    public void purge(String name) {
+        ensureOpen();
+        validateChildName(name);
+        final File removed = new File(rootDir, name + SUFFIX_REMOVED);
+        if (!removed.isDirectory()) {
+            throw newStorageNotFoundException(name);
+        }
+        try {
+            Util.deleteFileTree(removed);
```

Comment:
Deleting a file tree can be a time-consuming task. A client who issues this request might observe a timeout. How about renaming `*.removed` directory to `*.purged` and scheduling deletion in the background? This change will require some additional changes to your `StoragePurgeScheduler` such as moving the actual file deletion there.

Reply:
fixed it :-)